solr/core/src/java/org/apache/solr/cloud/ReplicateFromLeader.java

```diff
@@ -74,14 +74,18 @@ public void startReplication(boolean switchTransactionLog) {
               "SolrCore not found:" + coreName + " in " + CloudUtil.getLoadedCoreNamesAsString(cc));
         }
       }
+
       SolrConfig.UpdateHandlerInfo uinfo = core.getSolrConfig().getUpdateHandlerInfo();
+      String customPollInterval =
+          core.getSolrConfig().get("updateHandler").get("commitPollInterval").txt();
       String pollIntervalStr = "00:00:03";
+      String calculatedPollIntervalString = determinePollInterval(uinfo);
+
       if (System.getProperty("jetty.testMode") != null) {
         pollIntervalStr = "00:00:01";
-      }
-
-      String calculatedPollIntervalString = determinePollInterval(uinfo);
-      if (calculatedPollIntervalString != null) {
+      } else if (customPollInterval != null) {
+        pollIntervalStr = customPollInterval;
+      } else if (calculatedPollIntervalString != null) {
         pollIntervalStr = calculatedPollIntervalString;
       }
       log.info("Will start replication from leader with poll interval: {}", pollIntervalStr);
```

Comment:
logging: wondering about somewhere (not necessarily here) recording the custom interval and what it would have been otherwise.